pytensor/link/pytorch/linker.py

```diff
@@ -25,7 +14,16 @@ def fgraph_convert(self, fgraph, input_storage, storage_map, **kwargs):
     def jit_compile(self, fn):
         import torch
 
-        return torch.compile(fn)
+        from pytensor.link.pytorch.dispatch import pytorch_typify
+
+        inner_fn = torch.compile(fn)
+
+        # Torch does not accept numpy inputs and may return GPU objects
+        def fn(*inputs, inner_fn=inner_fn):
```

Comment:
Maybe just use the closure scoped function?

Reply:
Wdym?

Reply:
I think I usually see us just reference `inner_fn` without declaring an optional param. I think I've seen both but whatever works.